addons/html_builder/static/src/plugins/top_menu_visibility_option_plugin.js

```diff
@@ -0,0 +1,126 @@
+import { withSequence } from "@html_editor/utils/resource";
+import { registry } from "@web/core/registry";
+import { rgbToHex } from "@web/core/utils/colors";
+
+class TopMenuVisibilityOptionPlugin extends Plugin {
+    static id = "topMenuVisibilityOption";
+    resources = {
+        builder_actions: this.getActions(),
+        builder_options: [
+            withSequence(10, {
+                template: "html_builder.TopMenuVisibilityOption",
+                selector: "[data-main-object^='website.page('] #wrapwrap > header",
+            }),
+        ],
+        save_handlers: this.onSave.bind(this),
+    };
+
+    getActions() {
+        return {
+            setVisibility: {
+                clean: ({ editingElement, value: headerPositionValue }) => {
+                    callback[headerPositionValue](this.document, false);
+                },
+                apply: ({ editingElement, value: headerPositionValue }) => {
+                    callback[headerPositionValue](this.document, true);
+                    this.isDirty = true;
+                },
+                isApplied: ({ editingElement, value }) => this.getVisibilityItem() === value,
+            },
+            setIsDirty: {
+                apply: ({ editingElement }) => {
+                    this.isDirty = true;
+                },
+            },
+        };
+    }
+
+    getVisibilityItem() {
+        const isHidden = this.document
+            .querySelector("#wrapwrap > header")
+            .classList.contains("o_snippet_invisible");
+        const isOverlay = this.document
+            .getElementById("wrapwrap")
+            .classList.contains("o_header_overlay");
+        return isOverlay ? "overTheContent" : isHidden ? "hidden" : "regular";
+    }
+
+    getColorValue(attribute, classPrefix) {
+        const headerEl = this.document.querySelector("#wrapwrap > header");
+        const matchingClass = [...headerEl.classList].find((cls) => cls.startsWith(classPrefix));
+        return matchingClass || rgbToHex(headerEl.style.getPropertyValue(attribute));
+    }
+
+    onSave() {
+        if (!this.isDirty) {
+            return;
+        }
+        const item = this.getVisibilityItem();
+        const mainObject = this.services.website.currentWebsite.metadata.mainObject;
+        return Promise.all([
+            this.services.orm.write(mainObject.model, [mainObject.id], {
+                header_overlay: item === "overTheContent",
+            }),
+            this.services.orm.write(mainObject.model, [mainObject.id], {
+                header_visible: item !== "hidden",
+            }),
+            this.services.orm.write(mainObject.model, [mainObject.id], {
+                header_color: this.getColorValue("background-color", "bg-o-color-"),
+            }),
+            this.services.orm.write(mainObject.model, [mainObject.id], {
+                header_text_color: this.getColorValue("color", "text-o-color-"),
+            }),
```

Comment:
Can't we write all those fields with one `write`?
```suggestion
            this.services.orm.write(mainObject.model, [mainObject.id], {
                header_overlay: item === "overTheContent",
                header_visible: item !== "hidden",
                header_color: this.getColorValue("background-color", "bg-o-color-"),
                header_text_color: this.getColorValue("color", "text-o-color-"),
            }),

```